### ISSUE TITLE
error in inetmanet - wrong define LINUX

### ISSUE THREAD

**Issue by mfrey** [des-testbed/libara#132]
Compiling the current head of libARA using OMNeT++ 4.6 causes an error in the inetmanet submodule
```
linklayer/ext/ExtInterface.cc
linklayer/ext/ExtInterface.cc: In member function ‘virtual void ExtInterface::handleMessage(cMessage*)’:
linklayer/ext/ExtInterface.cc:144:18: error: ‘struct sockaddr_in’ has no member named ‘sin_len’
             addr.sin_len = sizeof(struct sockaddr_in);
                  ^
```
This causes another build error in ``util/headerserializers/headers/defs.h`` in line 64
``` 
  #elif defined(__linux__)
  # include <endian.h>
  # define LITTLE_ENDIAN __LITTLE_ENDIAN
  # define BIG_ENDIAN    __BIG_ENDIAN
  # define BYTE_ORDER    __BYTE_ORDER
  #elif defined(__APPLE__)
  #include <machine/endian.h>
  #else /* fallback, including cases __FreeBSD__, __NetBSD__ and __OpenBSD__ */
  //XXX this causes problems in FreeBSD, and probably not needed anyway: #define __BSD_VISIBLE
  #include <machine/endian.h>
  #endif
``` 
The actual issue is in line 55, but the compiler searches for ``endian.h`` in the wrong place.

**Comment by mfrey** [comment]
I'm not certain why this is not an issue with ``OMNeT++-4.3`` but ``OMNeT++-4.6``, but apparently the define ``linux`` is wrong, it has to be ``__linux__``.

**Comment by mfrey** [comment]
This issue has been fixed in [#2c99b22c36d2](https://github.com/des-testbed/inetmanet-2.0/commit/2c99b22c36d2dbdcfa5532c3088bd128dff16082) in the INETMANET repository.
